components/todo-input.js

```diff
@@ -1,20 +1,96 @@
-import React from "react";
-import styled from "styled-components";
+import React from 'react';
+import styled from 'styled-components';
 
-export default function TodoInput() {
-  return (
-    <Wrapper>
-      리액트-투두
-      <br />
-      시간을 입력하는 칸
-      <br />
-      Todo를 입력하는 칸
-    </Wrapper>
-  );
+export default function TodoInput(props) {
+	const { date, todo, onChangeTodo, onChangeDate, onSubmit } = props;
+
+	const maxLengthCheck = (object) => {
+		if (object.value.length > object.maxLength) {
+			object.value = object.value.slice(0, object.maxLength);
+		}
+	};
```

Comment:
```suggestion
export default function TodoInput(props) {
        const [ newTodo, setNewTodo ] = useState({ date : '', todo : ''});
	const { onSubmit } = props;

        const handleFormChange = (e) => {
           setNewTodo({...newTodo, [e.target.name]:e.target.value});
        };
```
1. props에서 ```date, todo, onChangeTodo, onChangeDate```를 받을 필요 없이,
이 컴포넌트 안에서 ```newTodo``` useState로 선언해주시고, ```setNewTodo``` 이용해 위에처럼 ```handleFormChange``` 이벤트 핸들러 작성해주시면 됩니다.
date, todo와 각각의 핸들러를 따로 작성해줄 필요 없이 위와 같이 코드 작성해주시면 됩니다.
2. ```pages/index.js```에서 작성하신 input value validate하는 if 조건이 처음에도 적용되게 하려면 ```newTodo```의 초깃값을 ```{ date : '', todo : ''}```로 설정해주시면 됩니다.